src/Effects/CombFilterBank.cpp

```diff
@@ -0,0 +1,125 @@
+#include <cassert>
+#include <cstdio>
+#include <cmath>
+#include <stdio.h>
+
+#include "../Misc/Allocator.h"
+#include "../Misc/Util.h"
+#include "CombFilterBank.h"
+
+namespace zyn {
+
+    CombFilterBank::CombFilterBank(Allocator *alloc, unsigned int samplerate_, int buffersize_):
+    inputgain(1.0f),
+    outgain(1.0f),
+    baseFreq(0.0f),
+    nrOfStrings(0),
+    memory(*alloc),
+    samplerate(samplerate_),
+    buffersize(buffersize_)
+    {
+        // setup the smoother for gain parameter
+        gain_smoothing.sample_rate(samplerate>>4);
+        gain_smoothing.thresh(0.02f); // TBD: 2% jump audible?
+        gain_smoothing.cutoff(1.0f);
+    }
+
+    CombFilterBank::~CombFilterBank()
+    {
+        setStrings(0, baseFreq);
+    }
+
+    void CombFilterBank::setStrings(unsigned int nrOfStringsNew, float baseFreqNew)
+    {
+        if(nrOfStringsNew == nrOfStrings && baseFreqNew == baseFreq) 
+            return;
+
+        const unsigned int mem_size_new = (int)ceilf(( (float)samplerate/baseFreqNew + buffersize + 2)/16) * 16;
+        if(mem_size_new == mem_size)
+        {
+            //limit to 88 Strings
+            nrOfStringsNew = (nrOfStringsNew>88) ? 88 : nrOfStringsNew;
+            if(nrOfStringsNew>nrOfStrings) 
+            {
+                for(unsigned int i = nrOfStrings; i < nrOfStringsNew; ++i)
+                {
+                    output[i] = (float*)memory.alloc_mem(mem_size*sizeof(float));
+                    memset(output[i], 0, mem_size*sizeof(float));
+                }
+            }
+            else if(nrOfStringsNew<nrOfStrings) 
+                for(unsigned int i = nrOfStringsNew; i < nrOfStrings; ++i)
+                    memory.dealloc(output[i]);
```

Comment:
==987872== Process terminating with default action of signal 11 (SIGSEGV)
==987872==  General Protection Fault
==987872==    at 0x379E13: block_is_free (tlsf.c:191)
==987872==    by 0x37BBE8: tlsf_free (tlsf.c:978)
==987872==    by 0x29D10D: zyn::AllocatorClass::dealloc_mem(void*) (Allocator.cpp:85)
==987872==    by 0x34D978: void zyn::Allocator::dealloc<float>(float*&) (Allocator.h:80)
==987872==    by 0x3C1302: zyn::CombFilterBank::setStrings(unsigned int, float) (CombFilterBank.cpp:52)